reward/src/lib.rs

```diff
@@ -0,0 +1,534 @@
+        term::blank();
+        term::info!("{}", term::format::tertiary("Sending transaction..."));
+        term::blank();
+        rt.block_on(term::ethereum::transaction(call))?;
+
+        // tag HEAD as last successful payout
+        if head_sha1.is_some() {
+            let spinner = term::spinner(&term::format::tertiary("Tagging HEAD as `rad/reward`"));
+            let head_commit = repo.find_commit(head_sha1.unwrap())?.into_object();
+            repo.tag_lightweight("rad/reward", &head_commit, true)?;
+            spinner.finish();
+        }
+    }
+
+    Ok(())
+}
+
+fn calculate_rewards(
+    strategy: String,
+    reward: U256,
+    all_ens: &[&radicle_common::person::Ens],
+) -> anyhow::Result<HashMap<String, U256>> {
+    let mut shares: HashMap<String, usize> = HashMap::new();
+    let _ = all_ens
+        .iter()
+        .map(|ens| {
+            if let Some(v) = shares.get_mut(&ens.name) {
+                *v += 1;
+            } else {
+                shares.insert(ens.name.clone(), 1);
+            }
+        })
+        .collect::<Vec<()>>();
+
+    let mut rewards: HashMap<String, U256> = HashMap::new();
+    if strategy == "equally" {
+        let count = U256::from(shares.len());
+        let each_reward = reward
+            .checked_div(count)
+            .ok_or_else(|| anyhow!("Couldn't divide total reward by number of contributors"))?;
+
+        let _ = shares
+            .keys()
+            .map(|ens| {
+                rewards.insert(ens.to_string(), each_reward);
+            })
+            .collect::<Vec<()>>();
+    } else if strategy == "weighted" {
+        let total_shares = shares.values().sum::<usize>();
+        let reward_per_share = reward.checked_div(U256::from(total_shares)).unwrap();
+        let _ = shares
+            .iter()
+            .map(|(ens, shares)| {
+                let this_reward = reward_per_share.checked_mul(U256::from(*shares)).unwrap();
+                rewards.insert(ens.to_string(), this_reward);
+            })
+            .collect::<Vec<()>>();
+    }
+
+    Ok(rewards)
+}
+
+fn amount_to_u256(amount: &str, decimals: usize) -> anyhow::Result<Option<U256>> {
+    let decimal_position = amount.chars().position(|c| c == '.');
+    if decimal_position.is_none() {
+        return Ok(U256::from_dec_str(amount)
+            .map_err(|_| anyhow!("Couldn't turn amount to U256"))?
+            .checked_mul(U256::exp10(decimals)));
+    }
+
+    let decimal_position = decimal_position.unwrap();
+    let mut fraction_length = amount.len() - decimal_position - 1;
+    let mut amount = amount.to_string();
+    amount.remove(decimal_position);
+
+    if decimals < fraction_length {
+        let extra_fraction_digits = fraction_length - decimals;
+        amount = amount[..amount.len() - extra_fraction_digits].to_string();
+        fraction_length -= extra_fraction_digits;
+    }
+
+    Ok(U256::from_dec_str(&amount)?.checked_mul(U256::exp10(decimals - fraction_length)))
+}
```

Comment:
Why is this function needed? Why is `U256::from_dec_str(&amount)` not enough?

Reply:
`U256::from_dec_str` deals with integers only. 

Here I parse human-readable numbers from (and to) string which is very important when moving dealing with money. Different ERC-20s have different decimals systems and it's super easy to mix up, and send ten times more reward.

Reply:
Ok I see, would be worth eventually (different PR) using quickcheck to do roundtrip testing of this and maybe move it to `common`

Reply:
Ok I see, would be worth eventually (different PR) using quickcheck to do roundtrip testing of this and maybe move it to `common`

Reply:
Interesting, will implement it as soon as this is merged.

Reply:
https://github.com/radicle-dev/radicle-cli/pull/124/commits/fe90454b59bb26e1369f2c8924e51782a4c9a200